### PR TITLE
enzyme 0.0.90

### PR DESCRIPTION
**Pull request by Porkepix**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.<details>
  <summary>release notes</summary>
  <pre>## What's Changed
* Support building in Yggdrasil with LLVM 16 by @vchuravy in https://github.com/EnzymeAD/Enzyme/pull/1508
* Add nicer error messages for sparse by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1509
* Update to LLVM main change by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1511
* Add metadata to specify whether an arg is overwritten by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1512
* Fix loop rematerialization [if allocation not in the loop] by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1513
* Fix memcpy derivative for any/float combo by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1514
* Fix blas byref alpha update size by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1515
* Fix inverted bundle when undef usage by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1516
* LLVM 17 by @tgymnich in https://github.com/EnzymeAD/Enzyme/pull/1387
* Fix macos test [blas] by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1518
* Eliminate warnings when building Enzyme by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1517
* Fix copy memory error [compile time] by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1519
* Mark isfinite as inactive by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1520
* Fix dlascl api by @wsmoses in https://github.com/EnzymeAD/Enzyme/pull/1522

**Full Changelog**: https://github.com/EnzymeAD/Enzyme/compare/v0.0.89...v0.0.90</pre>
</details>

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  ==> /opt/homebrew/opt/llvm/bin/opt input.ll --enable-new-pm=0 -load=/opt/homebrew/opt/enzyme/lib/LLVMEnzyme-17.dylib --enzyme-attributor=0 -enzyme -o output.ll -S
  opt: Unknown command line argument '--enable-new-pm=0'.  Try: '/opt/homebrew/opt/llvm/bin/opt --help'
  opt: Did you mean '--enable-newgvn=0'?
```

**Comment by wsmoses** [comment]
The test should probably be upgraded to just using `clang -fplugin=/path/to/ClangEnzyme-XX.dylib file.c -O1`.

**Comment by wsmoses** [comment]
I think this is due to LLVM dropping the old pm (and that test relied on it). The plugin directly will work for whatever pass manger (old or new), and also is a bit simpler/suggested way of using. Though the clang used needs to be the same one Enzyme was built against (so in this case the homebrew clang)

**Comment by chenrui333** [comment]
> The test should probably be upgraded to just using `clang -fplugin=/path/to/ClangEnzyme-XX.dylib file.c -O1`.

got this error

```
==> /opt/homebrew/opt/llvm/bin/clang input.ll -fplugin=/opt/homebrew/opt/enzyme/lib/LLVMEnzyme-17.dylib -enzyme -O1 -o test
ld: Undefined symbols:
  ___enzyme_autodiff, referenced from:
      _dsquare in input-f95e44.o
      _main in input-f95e44.o
  nzyme, referenced from:
      <initial-undefines>
```

**Comment by chenrui333** [comment]
- look at this pr, https://github.com/EnzymeAD/Enzyme/pull/1387, do I need ztsd?

**Comment by chenrui333** [comment]
actually I was using a wrong lib ref, fixing it now.

**Comment by wsmoses** [comment]
Or sorry the clang plugin applies to the C/C++ code, you shouldn't need to generate intermediate llvm (the clang plugin will register the llvm pass properly)

**Comment by chenrui333** [comment]
@wsmoses looks like it does not quite work for me

```
==> /opt/homebrew/opt/llvm/bin/clang input.ll -fplugin=/opt/homebrew/Cellar/enzyme/0.0.90/lib/ClangEnzyme-17.dylib -O1 -o test
ld: Undefined symbols:
  ___enzyme_autodiff, referenced from:
      _dsquare in input-2d1b51.o
      _main in input-2d1b51.o

```

File: Formula/e/enzyme.rb [hunk 1]
```diff
@@ -52,16 +50,11 @@ def install
...
-    # https://llvm.org/docs/OpaquePointers.html#version-support
     system ENV.cc, "-v", testpath/"test.c", "-S", "-emit-llvm", "-o", "input.ll", "-O2",
-                   "-fno-vectorize", "-fno-slp-vectorize", "-fno-unroll-loops",
-                   "-Xclang", "-no-opaque-pointers"
-    system opt, "input.ll", "--enable-new-pm=0",
-                "-load=#{opt_lib/shared_library("LLVMEnzyme-#{llvm.version.major}")}",
-                "--enzyme-attributor=0", "-enzyme", "-o", "output.ll", "-S"
-    system ENV.cc, "output.ll", "-O3", "-o", "test"
+                   "-fno-vectorize", "-fno-slp-vectorize", "-fno-unroll-loops"
+    system ENV.cc, "input.ll",
```
Comment (wsmoses) [inline Formula/e/enzyme.rb:55]:
```suggestion
    system ENV.cc, testpath/test.c,
```

File: Formula/e/enzyme.rb [hunk 2]
```diff
@@ -52,16 +50,11 @@ def install
     opt = llvm.opt_bin/"opt"
     ENV["CC"] = llvm.opt_bin/"clang"
 
-    # `-Xclang -no-opaque-pointers` is a transitional flag for LLVM 15, and will
-    # likely be need to removed in LLVM 16. See:
-    # https://llvm.org/docs/OpaquePointers.html#version-support
     system ENV.cc, "-v", testpath/"test.c", "-S", "-emit-llvm", "-o", "input.ll", "-O2",
```
Comment (wsmoses) [inline Formula/e/enzyme.rb:53]:
```suggestion
```

File: Formula/e/enzyme.rb [hunk 3]
```diff
@@ -52,16 +50,11 @@ def install
...
-                   "-fno-vectorize", "-fno-slp-vectorize", "-fno-unroll-loops",
-                   "-Xclang", "-no-opaque-pointers"
-    system opt, "input.ll", "--enable-new-pm=0",
-                "-load=#{opt_lib/shared_library("LLVMEnzyme-#{llvm.version.major}")}",
-                "--enzyme-attributor=0", "-enzyme", "-o", "output.ll", "-S"
-    system ENV.cc, "output.ll", "-O3", "-o", "test"
+                   "-fno-vectorize", "-fno-slp-vectorize", "-fno-unroll-loops"
```
Comment (wsmoses) [inline Formula/e/enzyme.rb:54]:
```suggestion
```

**Comment by chenrui333** [comment]
Thanks @wsmoses!

**Review by iMichka — APPROVED** [review]

**Pull request merged**